modules/auxiliary/gather/cloud_lookup.rb

```diff
@@ -0,0 +1,706 @@
+
+      unless host_header.eql? nil
+        http.set_config({'vhost' => host_header})
+      end
+
+      request = http.request_raw({
+        'method' => 'GET',
+        'uri' => uri,
+        'agent' => datastore['USERAGENT']
+      })
+      response = http.send_recv(request)
+      http.close
+
+    rescue ::Rex::ConnectionError, Errno::ECONNREFUSED, Errno::ETIMEDOUT, StandardError => error
+      print_error(error.message)
+    end
+    return false if response.nil?
+
+    return response
+  end
+
+  ## auxiliary/gather/censys_search.rb
+  def parse_ipv4(records)
+    ip_list = []
+    records.each do | ipv4 |
+      ip_list.push(ipv4['ip'])
+    end
+    return ip_list
+  end
+
+  ## auxiliary/gather/enum_dns.rb
+  def save_note(hostname, ip, port, sname, ebypass)
+    data = {'vhost' => hostname, 'real_ip' => ip, 'action' => @my_action.name, 'effective_bypass' => ebypass}
+    report_note(
+      :host => hostname,
+      :port => port,
+      :type => 'Cloud Lookup (and bypass)',
+      :data => data,
+      update: :unique_data
+    )
+  end
+
+  # ------------------------------------------------------------------------- #
+
+  def check_bypass(fingerprint, ip)
+    ret_value = false
+
+    # Check for "misconfigured" web server on TCP/80.
+    if(check_tcp_port(ip, 80))
+      found = check_request(fingerprint, ip, 80, false)
+    end
+    ret_value = true if found
+
+    # Check for "misconfigured" web server on TCP/443.
+    if(check_tcp_port(ip, 443))
+      found = check_request(fingerprint, ip, 443, true)
+    end
+    ret_value = true if found
+
+    return ret_value
+  end
+
+  def check_request(fingerprint, ip, port, ssl)
+    proto = (ssl ? 'https' : 'http')
+
+    vprint_status(" * Trying: #{proto}://#{ip}:#{port}/")
+    response = http_get_request_raw(ip, port, ssl, datastore['HOSTNAME'], datastore['URIPATH'])
+    if response != false
+      return false if detect_solution(response, @my_action.name)
+
+      if response.code.eql? 200
+        html = response.get_html_document
+        begin
+          if html.at(datastore['TAG']).to_s.include? fingerprint.to_s.encode('utf-8')
+            print_good("A direct-connect IP address was found: #{proto}://#{ip}:#{port}/")
+            save_note(datastore['HOSTNAME'], ip, port, proto, true)
+            return true
+          end
+        rescue NoMethodError, Encoding::CompatibilityError
+          return false
+        end
+      else
+        if response.redirect?
+          vprint_line("      --> responded with HTTP status code: #{response.code.to_s} to #{response.headers['location']}")
+          begin
+            if response.headers['location'].include?(datastore['hostname'])
+              print_warning("A leaked IP address was found: #{proto}://#{ip}:#{port}/")
+              save_note(datastore['HOSTNAME'], ip, port, proto, false) if datastore['REPORT_LEAKS'].eql? true
+            end
+          rescue NoMethodError, Encoding::CompatibilityError
```

Comment:
Ok, good remark, I need to protect the if condition `if html.at(datastore['TAG']).to_s.include? fingerprint.to_s.encode('utf-8')` because in rarely case it cause an error.

As far as I can remember, I chose to leave `save_note` in the block because I thought it was unlikely that it would generate an error.

What do you think?

Also I can test by adding a variable and by shifting `save_note`?

Reply:
I modified but I ask your opinion?